parser-typechecker/src/Unison/Codebase/Editor/HandleInput.hs

```diff
@@ -388,8 +388,11 @@ loop = do
             (failed, failedDependents) <-
               getEndangeredDependents (eval . GetDependents) toDelete rootNames
             if failed == mempty then do
-              stepManyAt . fmap (BranchUtil.makeDeleteTermName resolvedPath) . toList $ tms
-              stepManyAt . fmap (BranchUtil.makeDeleteTypeName resolvedPath) . toList $ tys
+              let makeDeleteTermNames = fmap (BranchUtil.makeDeleteTermName resolvedPath) . toList $ tms
+              let makeDeleteTypeNames = fmap (BranchUtil.makeDeleteTypeName resolvedPath) . toList $ tys
+              stepManyAt (makeDeleteTermNames ++ makeDeleteTypeNames)
+              root'' <- use root
+              respond $ ShowDiff input (Branch.namesDiff root' root'')
```

Comment:
Is grabbing the new root and diffing expensive? I noticed it's done differently in `delete.namespace`:
```haskell
            let deletedNames =
                  Names.prefix0
                    (Path.toName' (Path.unsplit' p))
                    (Branch.toNames0 b)
                diff = Names3.diff0 deletedNames mempty
            respond $ ShowDiff input diff
```

Reply:
Guessing the latter was just reusing ShowDiff without thinking.  It shouldn’t be particularly expensive.